### PR TITLE
UP-4813: Recover MediatingFactoryBean for JNDI

### PR DESCRIPTION
**Pull request by bjagg**
https://issues.jasig.org/browse/UP-4813

##### Checklist

-   [x] the [individual contributor license agreement][] is signed
-   [x] commit message follows [commit guidelines][]
-   [x] documentation is changed or added
-   [ ] tests are included

##### Description of change

Recover class thought to be unused in previous cleanup effort. This class is used with JNDI configurations of datasource.

[individual contributor license agreement]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#individual-contributor-license-agreement
[commit guidelines]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#commit
[message properties]: https://github.com/Jasig/uPortal/tree/master/uportal-war/src/main/resources/properties/i18n
[WCAG 2.0 AA]: https://www.w3.org/WAI/WCAG20/quickref/?levels=aaa&technologies=smil%2Cpdf%2Cflash%2Csl

### CODE REVIEW

File: docs/configure-using-jndi.md [hunk 1]
```diff
@@ -63,13 +69,66 @@ The third (and last!) change is to add the reference to
...
+## Define uPortal DataSources
+
+Defining DataSources external to an application was one of the earliest
+uses of JNDI. uPortal has support for JNDI DataSources. In addition,
+a helper factory bean can be configured to use the first working bean.
+ This allows the definition of both a JNDI DataSource bean along with
+ a fallback bean that is defined with filter file values for testing.
+ 
+Example from datasourceContext.xml:
+```aidl
```
Comment (ChristianMurphy) [inline docs/configure-using-jndi.md]: Is this example in [AIDL](https://developer.android.com/guide/components/aidl.html) or in XML?

File: docs/configure-using-jndi.md [hunk 2]
```diff
@@ -1,6 +1,12 @@
-# Use JNDI to override values for some properties
+# JNDI Usage in uPortal
+JNDI can be used to populate values in various areas of uPortal configuration.
```
Comment (ChristianMurphy) [inline docs/configure-using-jndi.md]: Could an empty newline be added between the heading and the paragraph?

File: docs/configure-using-jndi.md [hunk 3]
```diff
@@ -63,13 +69,66 @@ The third (and last!) change is to add the reference to
...
+uses of JNDI. uPortal has support for JNDI DataSources. In addition,
+a helper factory bean can be configured to use the first working bean.
+ This allows the definition of both a JNDI DataSource bean along with
+ a fallback bean that is defined with filter file values for testing.
+ 
+Example from datasourceContext.xml:
+```aidl
+
+    <!--
+    <bean id="PortalDb" parent="basePooledDataSource">
```
Comment (ChristianMurphy) [inline docs/configure-using-jndi.md]: Is starting the indentation at 4 spaces still meaningful with the surrounding code fence?

File: docs/configure-using-jndi.md (hunk 1 above)
Comment (ChristianMurphy) [inline docs/configure-using-jndi.md]: Could an empty newline be added between the paragraph and the code fence?

File: docs/configure-using-jndi.md (hunk 2 above)
Comment (bjagg) [inline docs/configure-using-jndi.md]: No problem

File: docs/configure-using-jndi.md (hunk 1 above)
Comment (bjagg) [inline docs/configure-using-jndi.md]: XML, will correct

Comment (bjagg) [inline docs/configure-using-jndi.md]: Sure

File: docs/configure-using-jndi.md (hunk 3 above)
Comment (bjagg) [inline docs/configure-using-jndi.md]: Yes, for someone cutting and pasting, the indentation will match the existing file.

**Pull request merged**